src/python/arcor2_kinect_azure_data/object_types/kinect.py

```
@@ -0,0 +1,182 @@
+    pass
+
+
+class KinectCamera(Camera):
+
+    _ABSTRACT = False
+
+    def __init__(
+        self,
+        obj_id: str,
+        name: str,
+        pose: Pose,
+        collision_model: Models,
+        url: str,
+        comm_timeout: float = 3,
+        depth_num_frames: int = 1,
+    ) -> None:
+        # super().__init__(obj_id, name, pose, collision_model) TODO: uncomment when finalizing
+
+        self._url = url
+        self._comm_timeout = comm_timeout
+
+        self._started: bool = False
+
+        self._settings: Optional[Settings] = None  # type: ignore
+        self._config: Optional[Config] = None
+
+        self._depth_num_frames = depth_num_frames
+
+    def _construct_path(self, path: str) -> str:
+        parsed = urlparse(self._url)
+        if parsed.netloc == "":
+            parsed = urlparse("http://" + self._url)
+        return urljoin(parsed.geturl(), path)
+
+    def _communicate_dataclass(
+        self,
+        relative_path: str,
+        out_dataclass: Type[DataClass],
+        body: Optional[DataClass] = None,
+        method: Method = Method.GET,
+    ) -> Optional[DataClass]:
+        full_path = self._construct_path(relative_path)
+        try:
+            return call(method, full_path, body=body, return_type=out_dataclass)
+        except RestHttpException:
+            return None
+
+    def _communicate_image(self, relative_path: str) -> Image.Image:
+        full_path = self._construct_path(relative_path)
+        return get_image(full_path)
+
+    def _set_position(self, position: Position) -> None:
+        ret_pos = self._communicate_dataclass(
+            "/position/set", out_dataclass=Position, body=Pose(position=position), method=Method.POST,
+        )
+        assert ret_pos == position
+
+    def start(self) -> None:
+        if self._started:
+            return
+
+        try:
+            self._communicate_dataclass("state/start", out_dataclass=KinectResponseModel)
+        except RestHttpException:
```

Isn't the exception swallowed by `_communicate_dataclass`? Btw, I would think that removing that swallowing and returning the non-optional type would be more clear.